core/src/main/java/org/modelmapper/spi/TypeMapMapping.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modelmapper.spi;
+
+import org.modelmapper.Condition;
+import org.modelmapper.Converter;
+import org.modelmapper.Provider;
+
+/**
+ * Represents mapping operations.
+ *
+ * @param <S> source type
+ * @param <D> destination type
+ *
+ * @author Chun Han Hsiao
+ */
+public interface TypeMapMapping<S, D> extends PropertyReferenceMapping<S, D> {
```

Comment:
This interface I think would make sense to name `PropertyMapper` since it's something that is used to create mappings.